modules/apm/src/main/java/org/elasticsearch/telemetry/apm/APMMeterRegistry.java

```diff
@@ -99,6 +105,16 @@ public DoubleGauge getDoubleGauge(String name) {
         return doubleGauges.get(name);
     }
 
+    public DoubleGaugeObserver registerDoubleGaugeObserver(String name, String description, String unit) {
```

Comment:
Can we flip this around so the impl is passed to the registration method? That is, pass a Supplier, and return nothing.

Reply:
Yes, it would mean waiting until inside an instance to do the registration, which was my concern.

Reply:
Changed.